Client Database System/Client Database System/Product and Services.cpp

```diff
@@ -8,13 +8,74 @@
 
 #include "Product and Services.hpp"
 #include <iostream>
+#include <fstream>
+#include <iomanip>
+#include <string>
 #include <vector>
 
 using namespace std;
 
-void AddPrice(vector<int>& priceVector){
+Products_and_Services::Products_and_Services(){
+    product = "No product";
+    serviceDescription = "No service description";
+}
+
+void Product_and_Services::AddPrice(vector<int>& priceVector){
     int userPrice;
     
     cin >> userPrice;
     priceVector.push_back(userPrice);
 }
+
+void Product_and_Services::ReadFromFile(vector<string>& productVector, vector<string>& descriptionVector){
+}
+
+void Product_and_Services::ListServices(){
+}
+
+void Product_and_Services::AddNewService(string product, string serviceDescription, vector<string>& productVector, vector<string>& descriptionVector){
+    cout << "Input product, service description: ";
+    cin >> product;
+    cin >> serviceDescription;
+    
+    productVector.push_back(product);
+    descriptionVector.push_back(serviceDescription);
+}
+
+void Product_and_Services::UpdateService(string product, string serviceDescription, vector<string>& productVector, vector<string>& descriptionVector){
+    string userProduct;
+    string userService;
+    
+    cin >> userProduct;
+    cin >> userService;
+    
+    unsigned int i;
+    for(i = 0; i < productVector.size(); ++i){
+        if(userProduct == productVector.at(i)){
+            cout << "Input new product: " << endl;
+            cin >> userProduct;
+        }
+    }
+    for(i = 0; i < descriptionVector.size(); ++i){
+        if(userService == descriptionVector.at(i)){
+            cout << "Input new service: " << endl;
+            cin >> userService;
+        }
```

Comment:
This reads the users input but we have to add them to the vector, and we also have to erase the old elements. We can do this with the `insert()` and `erase()` functions.

So, in this case, once you find the product you can get the input and use `productVector.insert(productVector.begin() + i , userProduct);` 
The first part "productVector.begin()" tells the insert to put userProduct at element 0. "+ i" moves the index number from zero to whatever number "i" is in the loop. Then "userProduct" tells insert() to insert "userProduct" before the the index.

erase() would be written similarly to insert(). For the productVector, it would be `productVector.erase(productVector.begin() + i + 1);`
The only difference here is that it has no variable to input. Instead it finds the index that you want to modify. Like here `productVector.begin()` starts the index at 0, then `+ i + 1` changes the index number to whatever i is in the loop plus 1 because we want to erase the old element, which was pushed up by 1 from the insert function.\